### PR TITLE
Gsw 677 fix calculation logic of the swap router

### PR DESCRIPTION
**Pull request by jinoosss**

### CODE REVIEW

File: packages/swap-router/src/swap-router/swap-router.ts [hunk 1]
```diff
@@ -21,7 +32,63 @@ export class SwapRouter {
...
+
+    const topByDirectSwapPool = availPools
+      .filter(
+        p =>
+          [inputTokenPath, outputTokenPath].includes(p.tokenAPath) ||
+          [inputTokenPath, outputTokenPath].includes(p.tokenBPath),
+      )
+      .filter((_, index) => index < TOP_N_DIRECT_SWAPS);
+
+    const topByTVL = availPools
```
Comment (r3v4s) [inline packages/swap-router/src/swap-router/swap-router.ts:78]: TVL should be aware of USD price in future

File: packages/swap-router/src/swap-router/swap-router.ts [hunk 2]
```diff
@@ -2,9 +2,20 @@ import BigNumber from "bignumber.js";
...
 import { EstimatedRoute, Route, RouteWithQuote } from "./swap-router.types";
 import { makeRouteKey } from "./utility/route.util";
+import { MAX_UINT64 } from "../constants";
+
+export const TOP_N = 10;
+export const TOP_N_DIRECT_SWAPS = 2;
+export const TOP_N_TOKEN_IN_OUT = 2;
+export const TOP_N_SECOND_HOP = 1;
+export const TOP_N_WITH_EACH_BASE_TOKEN = 3;
+export const TOP_N_WITH_BASE_TOKEN = 3;
```
[the remark this answers is not in the document]
Comment (jinoosss) [reply packages/swap-router/src/swap-router/swap-router.ts:18]: https://github.com/Uniswap/smart-order-router/blob/main/src/routers/alpha-router/config.ts

File: packages/swap-router/src/swap-router/swap-router.ts [hunk 3]
```diff
@@ -21,7 +32,63 @@ export class SwapRouter {
...
+    const availPools = pools.filter(pool => pool.liquidity > 0);
+    const topN = 10;
+
+    const topByBaseWithTokenIn = availPools
+      .filter(
+        p => p.tokenAPath === inputTokenPath || p.tokenBPath === inputTokenPath,
+      )
+      .sort((p1, p2) => {
+        const priceOfP1 =
+          p1.tokenAPath === inputTokenPath ? p1.price : 1 / p1.price;
```
Comment (r3v4s) [inline packages/swap-router/src/swap-router/swap-router.ts:49]: prefer ratio

File: packages/swap-router/src/swap-router/swap-router.ts (hunk 2 above)
Comment (r3v4s) [inline packages/swap-router/src/swap-router/swap-router.ts:18]: let's use default const from [here](https://github.com/Uniswap/smart-order-router/blob/db32bb1ba5e9cd8d525cd023934448b98d8df33d/src/routers/alpha-router/config.ts#L80-L87)

**Review by r3v4s — CHANGES_REQUESTED** [review]
few comments left.

one need to change => change const number to same as uniswap's default chain-id's config

**Review by r3v4s — DISMISSED** [review]

**Review by r3v4s — APPROVED** [review]

**Pull request merged**